tests/lib/json/src/main.c

```diff
@@ -39,6 +39,18 @@ struct obj_array {
 	size_t num_elements;
 };
 
+struct nested_elt_arr {
+	char *name[10];
+	size_t name_num;
+	int height[10];
+	size_t height_num;
+};
+
+struct obj_nested_arr {
+	struct nested_elt_arr elements[10];
+	size_t num_elements;
+};
```

Comment:
@kowalewskijan Did you check the functionality with different array sizes? You choose all with size 10 here.

Reply:
yes, I also ran it with `name[8]`, `height[7]`, `elements[5]`. However it shouldn't matter because I initialized only part of these arrays as stated [here](https://github.com/zephyrproject-rtos/zephyr/pull/38888/files#diff-d9940077aa34fe98ce8846e44c2ef770c149b21310a2646ed3fa261f8ea2a9c6R368-R382)

Reply:
I tested your implementation with following structure:
```
struct NameArray{
	char * name[5];
	size_t nameCnt;
} ;

struct NumberArray{
	char * number[4];
	size_t numberCnt;
} ;

struct Data{
	struct NameArray namePkg[3];
	size_t namePkgCnt;
	struct NumberArray numberPkg[6];
	size_t numberPkgCnt;
};
```
following error message:
```
E: Page fault at address (nil) (error code 0x0)
E: Linear address not present in page tables
E: Access violation: supervisor thread not allowed to read
E: PTE: not present
E: RAX: 0x0000000000000000 RBX: 0x0000000000102fbe RCX: 0x0000000000116522 RDX: 0x0000000000000001
E: RSI: 0x000000000010c7b5 RDI: 0x00000000001166dc RBP: 0x0000000000116430 RSP: 0x0000000000116400
E:  R8: 0x00000000001166dc  R9: 0x0000000000116b18 R10: 0x0000000000000000 R11: 0x0000000000000000
E: R12: 0x00000000001165f8 R13: 0x0000000000000000 R14: 0x0000000000116b18 R15: 0x0000000000000002
E: RSP: 0x0000000000116400 RFLAGS: 0x0000000000000246 CS: 0x0018 CR3: 0x000000000012f000
E: RIP: 0x00000000001030e6
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x10fe80 (test_encode)
E: Halting system
```

Reply:
Thanks for testing it. I'll try to reproduce that and see what it's happening.

Reply:
I think it's a problem with the offset parameter in the json_obj_decr, it is used differently for array and object.
Let me know if you need all of my test code. Thank you too.

Reply:
I tried to reproduce your issue with similar structures:
```
   struct nested_name_arr {
       char *name[9];
       size_t name_num;
   };
   
   struct nested_name2_arr {
       char *name2[19];
       size_t name2_num;
   };
   
   struct nested_height_arr {
       int height[7];
       size_t height_num;
   };
   
   struct obj_nested_arr {
       struct nested_name_arr name_elements[5];
       size_t num_name_elements;
       struct nested_name2_arr name2_elements[6];
       size_t num_name2_elements;
       struct nested_height_arr height_elements[4];
       size_t num_height_elements;
   };
```
But I couldn't get this issue on my STM32F030R8 board.
Please double check if you have used `*_NESTED` macro only on primitives array json_descr.
You should have descrs as follows:
```
   static const struct json_obj_descr NameArray_descr[] = {
       JSON_OBJ_DESCR_ARRAY_NESTED(struct NameArray, name,
                   5, nameCnt, JSON_TOK_STRING),
   };
   
    static const struct json_obj_descr NumberArray_descr[] = {
       JSON_OBJ_DESCR_ARRAY_NESTED(struct NumberArray, number,
                   4, numberCnt, JSON_TOK_STRING),
   };
   
   static const struct json_obj_descr Data_descr[] = {
       JSON_OBJ_DESCR_OBJ_ARRAY(struct Data, namePkg,
               3, namePkgCnt, NameArray_descr,
               ARRAY_SIZE(NameArray_descr)),
       JSON_OBJ_DESCR_OBJ_ARRAY(struct Data, numberPkg,
               6, numberPkgCnt, NumberArray_descr,
               ARRAY_SIZE(NumberArray_descr)),
   };
```
Also if you're using the `test_json_obj_nested_arr_encoding` function please make sure you used `Data_descr` in `json_obj_encode_buf` otherwise you can get a fault indeed when you try to encode data structure with wrong json_descr.

Reply:
I have found the mistake I have made. It was in the data to parse. Sorry for that.
Nice work, well done!